### DISCUSSION TITLE
Cannot read properties of undefined (reading 'providers')

### DISCUSSION THREAD

**Discussion by Teravla** [pocketbase/pocketbase#6148] [Q&A]
Hi everyone,

I’m using the CDN to import PocketBase in my front-end application. However, when I attempt to connect using the authWithOAuth2 method with the Discord provider, I encounter an error stating that the providers property is not found on RecordService.ts at line 593.

Since the providers property is part of the PocketBase code, I’m wondering if anyone has encountered this issue or has any ideas on how to resolve it. Any help would be greatly appreciated!

Thank you!

**Comment by ganigeorgiev** [comment]
Make sure that the PocketBase executable and the JS SDK versions are compatible with each other.

If you are using PocketBase v0.22.x (_you can find the version at the bottom-right corner in the dashboard_), you need to pin your JS SDK version to `<0.22.0` (aka. the max compatible is JS SDK v0.21.x).

If you are using PocketBase v0.23.x, you need to use JS SDK v0.22+.

You can refer to the [PocketBase v0.23.0 CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v0230) for details how to upgrade. 
Specifically for the JS SDK changes you can check [JS SDK v0.22.0 CHANGELOG](https://github.com/pocketbase/js-sdk/blob/master/CHANGELOG.md#0220).

If that doesn't help, please provide more details about the used JS SDK (aka. what is the CDN endpoint you are using) and the PocketBase executable version.

**Reply by Teravla** [reply]
Thanks! After several tests, I realized I was using an old version. After updating to 23.3, everything works well.

**Reply by gtedavid** [reply]
Told you you needed to be in a more recent version 😉

**Comment by Teravla** [comment]
@ganigeorgiev 
I'd like to take this opportunity to congratulate you on your work on this project, which I'm particularly fond of. Maybe this isn't the right place for it, but well done.
